### ISSUE TITLE
Dicom regression: segmentation fault

### ISSUE THREAD

**Issue by CGDogan** [openslide/openslide#494]
### Operating system

doesnt matter

### Platform

doesnt matter

### OpenSlide version

newest trunk as of 5 sept

### Slide format

dcm

### Issue details

https://github.com/jcupitt/libdicom/blob/main/data/test_files/sm_image.dcm should not open with openslide, so it should fail detect_format as of https://github.com/openslide/openslide/pull/487/commits/c8649b45f47b429aba120cbcc7bcf1ef70a4f2f9 by @bgilbert. but it does not, rather it tries to open it and gives segfault. but meanwhile I think some renames are under progress so I mention @jcupitt as well.

```
C  [libdicom.so.1+0x7f14]  dcm_dataset_contains+0x0 <-- segfault in here
C  [libdicom.so.1+0xc034]  dcm_filehandle_get_metadata+0xd4
C  [libopenslide.so.0+0x26fa8]  dicom_file_new+0x174
C  [libopenslide.so.0+0x27a90]  dicom_open+0x120
C  [libopenslide.so.0+0x9a90]  openslide_open+0x130
```

You can do just:

```
#include <stdio.h>
#include <inttypes.h>
#include <openslide.h>

int main(int argc, char **argv) {
  openslide_t *osr = openslide_open(argv[1]);

  int64_t w, h;
  for (int i = 0; i < openslide_get_level_count(osr); i++) {
    openslide_get_level_dimensions(osr, i, &w, &h);
    printf("%d %"PRId64" %"PRId64"\n", i, w, h);
  }
}
```

**Comment by jcupitt** [comment]
Hi @CGDogan,

This is fixed with the libdicom that's proposed for 1.0:

```
$ openslide-show-properties sm_image.dcm 
dicom.AccessionNumber: 'S19-1_A_1_1'
dicom.AcquisitionDateTime: '20091229095915'
...
dicom.VolumetricProperties: 'VOLUME'
openslide.level-count: '1'
openslide.level[0].downsample: '1'
openslide.level[0].height: '50'
openslide.level[0].tile-height: '10'
openslide.level[0].tile-width: '10'
openslide.level[0].width: '50'
openslide.mpp-x: '0.499'
openslide.mpp-y: '0.499'
openslide.vendor: 'dicom'
```

**Comment by jcupitt** [comment]
Ah you might have been bitten by the rename -- openslide main needs a small patch (`dcm_filehandle_get_metadata()` needs changing to `dcm_filehandle_get_metadata_subset()`).

**Comment by bgilbert** [comment]
The change mentioned in https://github.com/openslide/openslide/issues/494#issuecomment-1706328094 landed in #498, and `sm_image.dcm` isn't crashing on my system.  I'll go ahead and close this out, but feel free to reopen if you're still seeing problems.  Thanks for the report!
